src/fastNLOConstants.h

```diff
@@ -0,0 +1,193 @@
+#ifndef __fnloconstants__
+#define __fnloconstants__
+
+#include <string>
+#include <vector>
+#include <set>
+
+#ifndef FNLO_NAME
+#define FNLO_NAME       "fastNLO_toolkit"
+#define FNLO_SUBPROJECT "toolkit"
+#define FNLO_VERSION    "2.3.1pre"
+#define FNLO_SVNREV     "2411"
+#define FNLO_AUTHORS    "D. Britzger, T. Kluge, K. Rabbertz, F. Stober, G. Sieber, M. Wobisch"
+#define FNLO_WEBPAGE    "http://projects.hepforge.org/fastnlo"
+#define FNLO_AUTHORSv14 "T. Kluge, K. Rabbertz, M. Wobisch"
+#define FNLO_QUOTEv14   "hep-ph/0609285"
+#define FNLO_AUTHORSv2  "D. Britzger, T. Kluge, K. Rabbertz, F. Stober, M. Wobisch"
+#define FNLO_QUOTEv2    "arXiv:1109.1310"
+#define FNLO_YEARS      "2005-2019"
+#endif
+
+// Define variables fixed by precompiler for optional parts.
+// Compilers like gcc usually drop unreachable statements
+// anyway such that unnecessary if-else-endif calls or
+// "unresolved symbol" errors are avoided.
+#define FNLO_HOPPET     ""
+#define FNLO_QCDNUM     ""
+#define FNLO_ROOT       "@ROOT@"
+
+// KR: Replace by precompiler defines
+const double TWOPI    =  6.28318530717958647692528;
+const double TWOPISQR = 39.47841760435743447533796;
+const double TOCL90   =  1.64485362695147271486385; // SQRT(2.D0)*InvERF(0.9D0)
+//#define TWOPI (2.*M_PI)
+//#define TWOPISQR (4.*M_PI*M_PI)
+// PDG values 2012/2013
+#define PDG_MU   (0.0023)
+#define PDG_MD   (0.0048)
+#define PDG_MS   (0.095)
+#define PDG_MC   (1.275)
+#define PDG_MB   (4.18)
+#define PDG_MT   (173.07)
+#define PDG_MZ   (91.1876)
+#define PDG_ASMZ (0.1184)
+
+namespace fastNLO {
+
+   // ---- typedefs ---- //
+   typedef std::vector<double > v1d;
+   typedef std::vector<std::vector<double > > v2d;
+   typedef std::vector<std::vector<std::vector<double > > > v3d;
+   typedef std::vector<std::vector<std::vector<std::vector<double > > > > v4d;
+   typedef std::vector<std::vector<std::vector<std::vector<std::vector<double > > > > > v5d;
+   typedef std::vector<std::vector<std::vector<std::vector<std::vector<std::vector<double > > > > > > v6d;
+   typedef std::vector<std::vector<std::vector<std::vector<std::vector<std::vector<std::vector<double > > > > > > > v7d;
```

Comment:
Oh, is that really necessary?

Reply:
I admit it's a bit ugly, but I left the original fastNLO headers as untouched
I've seen you have merged, thanks!